link.go

```diff
@@ -54,8 +54,8 @@ type link struct {
 
 // attach sends the Attach performative to establish the link with its parent session.
 // this is automatically called by the new*Link constructors.
-func (l *link) attach(ctx context.Context, s *Session, beforeAttach func(*frames.PerformAttach), afterAttach func(*frames.PerformAttach)) error {
-	if err := s.allocateHandle(l); err != nil {
+func (l *link) attach(ctx context.Context, beforeAttach func(*frames.PerformAttach), afterAttach func(*frames.PerformAttach)) error {
```

Comment:
Was this just a weird vestigial artifact that the we didn't use the session field? Or was there some utility in being able to force a particular session?

Reply:
Looks like this was cruft introduced in https://github.com/Azure/go-amqp/pull/165.